### PR TITLE
Adding cluster as an annotation instead of using ClusterName object meta

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

File: name.go [hunk 1]
```diff
@@ -65,12 +66,20 @@ func (n Name) String() string {
...
 type Object interface {
-	GetClusterName() string
+	GetAnnotations() map[string]string
+	SetAnnotations(annotations map[string]string)
 }
 
 // From returns the logical cluster name for obj.
 func From(obj Object) Name {
-	return Name{obj.GetClusterName()}
+	//TODO: Do we want some sort of error in this library when this is not found?
```
Comment (ncdc) [inline name.go:75]: I don't think it's necessary. We don't have it today. It would indicate a more fundamental problem with the storage plumbing.

File: name.go [hunk 2]
```diff
@@ -65,12 +66,20 @@ func (n Name) String() string {
...
+	return Name{obj.GetAnnotations()[LogicalClusterAnnotationKey]}
+}
+
+func (n Name) Set(obj Object) {
+	annotations := obj.GetAnnotations()
```
Comment (ncdc) [inline name.go:80]: if map nil, allocate

File: name_test.go [hunk 3]
```diff
@@ -68,3 +68,48 @@ func TestJSON(t *testing.T) {
...
+
+func TestSet(t *testing.T) {
+
+	testCases := []struct {
+		name          string
+		LogicaCluster Name
+		Object        *testObject
+	}{
+		{
+			name:          "Set cluster if not set",
```
Comment (ncdc) [inline name_test.go:92]: Add test case where annotations is nil

**Comment by ncdc** [comment]
I think we should defer merging this until we're ready to update kcp-dev/kubernetes and kcp, which will be after we cut 0.5. Ok to move to draft until then?

**Comment by shawn-hurley** [comment]
Converted to draft, will allocate if nil. Have a branch where I am using this PR and I cherry-picked the removal of the ClusterName to try and get it building and passing tests.

**Comment by ncdc** [comment]
Also this will need to be in a v2 package

File: name.go [hunk 4]
```diff
@@ -36,6 +36,7 @@ type Name struct {
 }
 
 const separator = ":"
+const LogicalClusterAnnotationKey = "tenancy.kcp.dev/cluster"
```
Comment (ncdc) [inline name.go:39]:
```suggestion
const AnnotationKey = "tenancy.kcp.dev/cluster"
```

Also, you can create a `( )` group and put both of these constants in it.

**Comment by ncdc** [comment]
Replaced by #21
